examples/starknet_hello_world/src/lib.cairo

```diff
@@ -1,58 +1,65 @@
-#[abi]
-trait IBalance {
-    fn increase_balance(amount: felt252);
-    fn get_balance() -> felt252;
+#[starknet::interface]
+trait IBalance<T> {
+    fn replace(ref self: T, a: u128) -> u128;
 }
 
-#[contract]
+#[starknet::contract]
 mod Balance {
+    use traits::Into;
+
+    #[storage]
     struct Storage {
-        balance: felt252, 
+        value: u128, 
     }
 
     #[constructor]
-    fn constructor(balance_: felt252) {
-        balance::write(balance_);
+    fn constructor(ref self: ContractState, value_: u128) {
+        self.value.write(value_);
     }
 
-    // Increases the balance by the given amount.
     #[external]
-    fn increase_balance(amount: felt252) {
-        balance::write(balance::read() + amount);
-    }
-
-    // Returns the current balance.
-    #[view]
-    fn get_balance() -> felt252 {
-        balance::read()
+    fn replace(ref self: ContractState, a: u128) -> u128 {
+        let value = self.value.read();
+        self.value.write(a);
+        value
```

Comment:
What about getting the current balance? I think we should still have a method for that, one that takes `self: @ContractState`

Reply:
I'm pretty sure external functions shouldn't be allowed to return values. Transactions to starknet don't have any return data anyway so not sure how that would work.